frappe/__init__.py

```diff
@@ -426,11 +426,19 @@ def has_website_permission(doctype, ptype="read", doc=None, user=None, verbose=F
 	if not user:
 		user = session.user
 
-	for method in (get_hooks("has_website_permission") or {}).get(doctype, []):
-		if not call(get_attr(method), doc=doc, ptype=ptype, user=user, verbose=verbose):
-			return False
+	hooks = (get_hooks("has_website_permission") or {}).get(doctype, [])
+	if hooks:
+		for method in hooks:
+			result = call(get_attr(method), doc=doc, ptype=ptype, user=user, verbose=verbose)
+			# if even a single permission check is Falsy
+			if not result:
+				return False
 
-	return True
+		# else it is Truthy
+		return True
+
+	else:
+		return False
 
```

Comment:
In summary:
1. if there are no hooks then no website permission
1. if even one hook returns false, then no permission
1. if all hooks return Truthy, then return True